src/GafferSceneUI/UVView.cpp

```diff
@@ -0,0 +1,912 @@
+				);
+			}
+			else if( output == texturesPlug() )
+			{
+				const boost::regex attrRegex( "<attr:([^>]+)>" );
+				ConstCompoundObjectPtr udimQuery = udimQueryPlug()->getValue();
+				const string textureFileName = textureFileNamePlug()->getValue();
+
+				CompoundObjectPtr result = new CompoundObject;
+				for( const auto &udim : udimQuery->members() )
+				{
+					set<string> udimFileNames;
+					const CompoundObject *objects = static_cast<const CompoundObject *>( udim.second.get() );
+					for( const auto &object : objects->members() )
+					{
+						string substitutedFileName = textureFileName;
+						const auto *attributes = static_cast<const CompoundObject *>( object.second.get() );
+						boost::sregex_iterator it( textureFileName.begin(), textureFileName.end(), attrRegex );
+						while( it != boost::sregex_iterator() )
+						{
+							string attribute;
+							if( auto *s = attributes->member<StringData>( it->str( 1 ) ) )
+							{
+								attribute = s->readable();
+							}
+							boost::replace_all( substitutedFileName, it->str( 0 ), attribute );
+							++it;
+						}
+
+						boost::replace_all( substitutedFileName, "<UDIM>", udim.first.string() );
+						udimFileNames.insert( substitutedFileName );
+					}
+
+					result->members()[udim.first] = new StringVectorData(
+						vector<string>( udimFileNames.begin(), udimFileNames.end() )
+					);
+				}
+				static_cast<CompoundObjectPlug *>( output )->setValue( result );
+			}
+			else
+			{
+				SceneProcessor::compute( output, context );
+			}
+		}
+
+	private :
+
+		StringVectorDataPlug *udimQueryPathsPlug()
+		{
+			return getChild<StringVectorDataPlug>( g_firstPlugIndex + 4 );
+		}
+
+		const StringVectorDataPlug *udimQueryPathsPlug() const
+		{
+			return getChild<StringVectorDataPlug>( g_firstPlugIndex + 4 );
+		}
+
+		StringPlug *udimQueryAttributesPlug()
+		{
+			return getChild<StringPlug>( g_firstPlugIndex + 5 );
+		}
+
+		const StringPlug *udimQueryAttributesPlug() const
+		{
+			return getChild<StringPlug>( g_firstPlugIndex + 5 );
+		}
+
+		CompoundObjectPlug *udimQueryPlug()
+		{
+			return getChild<CompoundObjectPlug>( g_firstPlugIndex + 6 );
+		}
+
+		const CompoundObjectPlug *udimQueryPlug() const
+		{
+			return getChild<CompoundObjectPlug>( g_firstPlugIndex + 6 );
+		}
+
+		StringVectorDataPlug *isolatePathsPlug()
+		{
+			return getChild<StringVectorDataPlug>( g_firstPlugIndex + 7 );
+		}
+
+		const StringVectorDataPlug *isolatePathsPlug() const
+		{
+			return getChild<StringVectorDataPlug>( g_firstPlugIndex + 7 );
+		}
+
+		static size_t g_firstPlugIndex;
+
+};
+
+IE_CORE_DEFINERUNTIMETYPED( UVView::UVScene );
+size_t UVView::UVScene::g_firstPlugIndex;
+
+//////////////////////////////////////////////////////////////////////////
+// GridGadget
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+class GridGadget : public GafferUI::Gadget
```

Comment:
Yeah, I wondered about this - perhaps in some general purpose GL-drawing code somewhere. There were some requirements here that meant I wasn't sure what it should look like - the labelling for UVs is fairly specific, and it also didn't seem to make sense to have more than one level of subdivisions. I decided to avoid generalising when I didn't know what general looked like yet.